### PR TITLE
Gatewayclass supportedfeatures metrics

### LINKED ISSUE #17: metrics: Add gatewayclass metrics for status.supportedFeatures
From GEP https://github.com/kubernetes-sigs/gateway-api/pull/2163/files, a new field in the GatewayClass status is proposed.
It looks something like this:

```yaml
  supportedFeatures:
    - HTTPRoute
    - HTTPRouteHostRewrite
    - HTTPRoutePortRedirect
    - HTTPRouteQueryParamMatching
```

It would be good to have this status information captured in state metrics.
e.g.

```prometheus
gatewayapi_gatewayclass_status_supported_feature{namespace="default",name="external",feature="HTTPRoute"} 1
gatewayapi_gatewayclass_status_supported_feature{namespace="default",name="external",feature="HTTPRouteHostRewrite"} 1
...
```

cc @LiorLieberman
Did I miss anything here from your suggestion on the sig call?

### PR DESCRIPTION
**Pull request by Ygnas**
Closes: #17

### CODE REVIEW

File: examples/kube-prometheus/kustomization.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ kind: Kustomization
 
 resources:
   - github.com/prometheus-operator/kube-prometheus?ref=release-0.11
-  - github.com/kuadrant/gateway-api-state-metrics?ref=0.1.0
+  - github.com/kuadrant/gateway-api-state-metrics?ref=0.1.1
```
Comment (david-martin) [inline examples/kube-prometheus/kustomization.yaml]: let's hold off on this bump for now

File: tests/e2e/main_test.go [hunk 2]
```diff
@@ -106,6 +106,29 @@ func testGatewayClasses(t *testing.T, metrics map[string][][]string) {
 	expectEqual(t, gatewayClass1Status1Labels["customresource_version"], "v1beta1", "gatewayapi_gatewayclass_status__1 customresource_version")
 	expectEqual(t, gatewayClass1Status1Labels["name"], "testgatewayclass1", "gatewayapi_gatewayclass_status__1 name")
 	expectEqual(t, gatewayClass1Status1Labels["type"], "Accepted", "gatewayapi_gatewayclass_status__1 type")
+
+	//gatewayapi_gatewayclass_status
```
Comment (david-martin) [inline tests/e2e/main_test.go]:
```suggestion
	//gatewayapi_gatewayclass_status_supported_features
```

File: config/gateway-api/crd/standard/gateway.networking.k8s.io_gatewayclasses.yaml [hunk 3]
```diff
@@ -234,6 +234,9 @@ spec:
         - jsonPath: .metadata.creationTimestamp
           name: Age
           type: date
+        - jsonPath: .status.supportedFeatures
+          name: SupportedFeatures
+          type: string
```
Comment (david-martin) [inline config/gateway-api/crd/standard/gateway.networking.k8s.io_gatewayclasses.yaml:239]: Is this change from the upstream gep too? 
Is it needed for the metrics piece?
Asking as I'm not sure what the impact is i.e. having the output be a string format when it's actually an  array of strings. Maybe that's ok?

Comment (Ygnas) [reply config/gateway-api/crd/standard/gateway.networking.k8s.io_gatewayclasses.yaml:239]: Oh yes forgot about this, it is not needed.

**Review by david-martin — APPROVED** [review]

**Pull request merged**